npm/extensions.bzl

```diff
@@ -53,7 +63,16 @@ def _extension_impl(module_ctx):
             if attr.npmrc:
                 npmrc = parse_npmrc(module_ctx.read(attr.npmrc))
                 (registries, npm_auth) = npm_translate_lock_helpers.get_npm_auth(npmrc, module_ctx.path(attr.npmrc), module_ctx.os.environ)
-            imports = npm_translate_lock_helpers.gen_npm_imports(importers, packages, lock_patched_dependencies, attr.pnpm_lock.package, attr.name, attr, registries, utils.default_registry(), npm_auth)
+
+            lifecycle_hooks, lifecycle_hooks_execution_requirements = macro_helpers.macro_lifecycle_args_to_rule_attrs(
+                attr.lifecycle_hooks,
+                attr.lifecycle_hooks_exclude,
+                attr.run_lifecycle_hooks,
+                attr.lifecycle_hooks_no_sandbox,
+                attr.lifecycle_hooks_execution_requirements,
+            )
+            imports = npm_translate_lock_helpers.gen_npm_imports(importers, packages, lock_patched_dependencies, attr.pnpm_lock.package, attr.name, attr, lifecycle_hooks, lifecycle_hooks_execution_requirements, registries, utils.default_registry(), npm_auth)
```

Comment:
eek, the order dependency and the number of args here is scary. perhaps switch to named parameters:
```
...gen_npm_imports(
   importers = importers,
   packages = packages,
   ...
```
?